### ISSUE TITLE
Allow sending decentralized pledges in plain-text

### ISSUE THREAD

**Issue by prusnak** [vinumeris/lighthouse#121]
When I do a decentralized pledge, Lighthouse generates a small (around 300 bytes) file, which should be sent via email to pledge organizer. 

Lighthouse can do a Base58 transformation of this file and this plain-text data can be sent via instant messaging. Organizer will then paste this info into Lighthouse resulting in pledge file to be created like it would be using email.

**Comment by mikehearn** [comment]
Heh. Might be useful but is a bit geeky! What's wrong with just sending the file via the IM app directly? Are you using a chat system that doesn't support sending files?

**Comment by prusnak** [comment]
Sending files via IM works in 50% of attempts. Usually because of NATs and/or IM client incompatibility.

**Comment by mikehearn** [comment]
I'm not against such a feature, but it needs some careful UI thought. As it only really exists to work around bugs in other apps it would probably need to be tucked out of the way a little bit. As is, you can always use email or other ways to move files around that are more reliable. I think commercial IM services tend to have centralised fallbacks for when UPnP and friends don't work, so I guess this is mostly an issue for IRC and Jabber users.

If anyone has ideas for a subtle UI for this feature let me know. Patches and contributions also welcome once we figure out the right UI.

**Comment by prusnak** [comment]
Right, agreed. Jabber and IRC is what geeks use. :-)

**Comment by slush0** [comment]
Actually text-based pledges may have potential to go viral. From psychological standpoint it's completely different if campaign on Facebook tells  "support us and send your pledge to email@example.com" or "support us and put pledges to comments" :-).

By the way, a related thing for which I don't want to create separate issue: Please make it clear in UI that such pledge file does not contain any secret information. One guy already sent me pledge file encrypted by GPG and I think this will be common concern of many...

**Comment by mikehearn** [comment]
Good point about virality. I had not thought of that angle. For my own projects I'm using a server, so I don't need this myself. I'm hoping someone will appear and start contributing code, for things like this, as I don't plan to work on Lighthouse forever.

Currently secretness of pledges is a bit of a mess. The email address field is only private for server assisted projects. So if pledges were to be posted on a web forum then the emails and comments would be visible (the ui does say this though). A better way would be to encrypt any sensitive data in pledges (I think this only means comments/names/emails) using a project key.

**Comment by slush0** [comment]
Ok, should I create separate issue for that?

I would like to contribute to project, but Java is "no go" for me. Maybe I'll create pledges for such features in Lighthouse :-)).

**Comment by mikehearn** [comment]
I'll probably switch the code base to Kotlin at some point, to get a better type system etc.

Yes a separate issue would be good, thanks.

**Comment by dabura667** [comment]
+1

Am currently trying to conceptualize ways the "Decentralized mode" can work in a way that informs everyone of the current pledge amount...

Maybe a bot that reads comments for the text-based pledges and adds them up, updating the total pledged with each comment...

Man, reddit just got more interesting...

**Comment by mikehearn** [comment]
If you have a bot you might as well have a server. The point of serverless mode is it doesn't rely on any third party (bitcoin/lighthouse) specific infrastructure.

**Comment by dabura667** [comment]
Am I correct in the assumption that if I just put up a project file and receive pledges via e-mail, each pledger will not know how much is left to be pledged unless I tell them explicitly?

The project file doesn't store any info about how many / how much pledges have been made, correct?

**Comment by mikehearn** [comment]
That's right. You have to coordinate that in some other way. Project files are just .... files .... so they can't change themselves to say how much money was raised. It's the big downside of the entirely serverless approach.

For most use cases, a simple fix is to mandate that a pledge be a multiple of a certain size. For instance you can say you want to raise 1 BTC and pledges must be multiples of 0.1 BTC with a max pledge size of 0.1 BTC (i.e. you have to pledge that much). Now you just count up the number of visible pledges to know how much money has been raised (ignoring revocations, of course, which complicates things further).

**Comment by dabura667** [comment]
hmmmmmmm... seems rough... 

I'll keep testing and playing around with ideas and report any interesting use cases I can find.
